packages/@jsii/go-runtime-test/project/runtime_type_checking_test.go

```diff
@@ -0,0 +1,63 @@
+package tests
+
+import (
+	"fmt"
+	"github.com/aws/jsii-runtime-go"
+	"github.com/aws/jsii/jsii-calc/go/jsiicalc/v3"
+	"github.com/aws/jsii/jsii-calc/go/jsiicalc/v3/anonymous"
+	"testing"
+)
+
+func TestConstructor(t *testing.T) {
```

Comment:
If we're going to add the nested type checking to .NET and Java, we should add it to Go as well.